### PR TITLE
Gifs Spice: small code cleanup, ready for launch

### PR DESCRIPTION
**Pull request by moollaza**
This one should be ready now for production. I'm wondering if we should rename the package `DDG::Spice::Giphy` or potentially, `DDG::Spice::Gifphy::Search` -- I'd like to make IA's for their other endpoints so the names would be similar, i.e. `Gihpy::Trending`, `Giphy::Random`, `Giphy::Translate`

I also noticed that our CSS isn't applying the `margin: auto` to the images in the detail pane so they're floating off to the left. Not sure if we want to fix that internally or add CSS to this IA?

2015-04-12_19h48_50

---

I noticed that on mobile, the detail view uses the fixed image instead of the Gif -- is that expected?

**Update:** Looking at the template, we use the thumbnail for the detail view on mobile (because we set `loadHighRes = false` -- I think we need better handling of that for gifs? No point it showing a static image in the detail, even on mobile...

/cc @jagtalon @bsstoner @russellholt 

---

> IA Page: https://duck.co/ia/view/gifs

### CODE REVIEW

File: share/spice/gifs/gifs.js [hunk 1]
```diff
@@ -1,31 +1,43 @@
...
-        },
-        meta: {
-            sourceName: 'Giphy',
-            sourceUrl: 'http://giphy.com/search/' + encodeURIComponent(searchTerm),
-            sourceIcon: true,
-            count: res.pagination.count,
-            total: res.pagination.total_count,
-            itemType: 'Gifs'
-        },
+        var searchTerm = DDG.get_query().replace(/gifs?/i,'').trim();
```
Comment (MrChrisW) [inline share/spice/gifs/gifs.js]: This won't match the trigger `giphy`

Comment (moollaza) [inline share/spice/gifs/gifs.js]: :+1:

File: share/spice/gifs/gifs.js [hunk 2]
```diff
@@ -1,31 +1,43 @@
...
-        },
-        meta: {
-            sourceName: 'Giphy',
-            sourceUrl: 'http://giphy.com/search/' + encodeURIComponent(searchTerm),
-            sourceIcon: true,
-            count: res.pagination.count,
-            total: res.pagination.total_count,
-            itemType: 'Gifs'
-        },
+        var searchTerm = DDG.get_query().replace(/gifs?|giphy/ig,'').trim();
```
Comment (jagtalon) [inline share/spice/gifs/gifs.js]: Maybe shouldn't be `//g`? There's the case where you could search for "giphy gifs" and it would strip out both of them.

Comment (jagtalon) [inline share/spice/gifs/gifs.js]: `trim()` looks like it's only on IE 9. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim

**Comment by jagtalon** [comment]
I actually like the gifs to move in the thumbnails but they eat up too much data. Got 17mb for "hello gif". We should solve the mobile problem on our end though.

File: share/spice/gifs/gifs.js (hunk 2 above)
Comment (MrChrisW) [inline share/spice/gifs/gifs.js]: >trim() looks like it's only on IE 9. 

You deploy a polyfill on production for the `trim` function :+1:  
https://duckduckgo.com/d1758.js

Comment (jagtalon) [inline share/spice/gifs/gifs.js]: @MrChrisW I should know that--thanks!

**Comment by moollaza** [comment]
I think this one can move ahead then?

**Comment by jagtalon** [comment]
@moollaza I added `image_mobile` which shows the gif (instead of the thumbnail) on mobile. :+1:

**Pull request merged**